src/lib/schema/role-schema.ts

```diff
@@ -3,9 +3,11 @@ import joi from 'joi';
 export const permissionRoleSchema = joi
     .object()
     .keys({
-        id: joi.number().required(),
+        id: joi.number(),
+        name: joi.string(),
```

Comment:
This validation now expects that either you provide the id or the name (or both)